kvserver: store.Enqueue shouldn't start verbose tracing

andrewbaptist opened the pull request:
Previously in server.Enqueue, we unconditionally started verbose
tracing. This is expensive and unnessary since most traces were never
processed. Additionally, with the new raft level logging for verbose
traces this resulted in unnecessary traces being created and logged to
the main cockroach.log. This change removes the automatic tracing inside
Enqueue and instead converts tests and callers to explicitly start a
span prior to calling the queue.

As part of this change the Enqueue method no longer returns a Recording.

Epic: none

Release note: None

Commit:
kvserver: store.Enqueue shouldn't start verbose tracing

Previously in server.Enqueue, we unconditionally started verbose
tracing. This is expensive and unnessary since most traces were never
processed. Additionally, with the new raft level logging for verbose
traces this resulted in unnecessary traces being created and logged to
the main cockroach.log. This change removes the automatic tracing inside
Enqueue and instead converts tests and callers to explicitly start a
span prior to calling the queue.

As part of this change the Enqueue method no longer returns a Recording.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132713)
<!-- Reviewable:end -->

andrewbaptist commented:
Making this change enables default raft logging as part of #131850.

andrewbaptist commented:
There are only 3 meaningful "non-test" changes, and only the first one changes any behavior.
* `replicate_queue.go` where we no longer trace every attempt. This changes behavior in that if the replicate queue is called without a tracing span, we might have less visibility into the underlying problem. However this is probably an acceptable tradeoff. This behavior was added in #86007.
* `split_queue.go` where we will only start a tracing span if log.ExpensiveLogEnabled is set. We would only log at the end if it was anyway, so this shouldn’t change things. I didn’t love how this was done, but didn’t want to increase scope by changing it. This won't change any behavior, but might still log more than we want.
* `store.go` - Enqueue no longer constructs a tracing span. Instead callers to it do. This won't change any behavior.

andrewbaptist commented:
I've simplified this PR to remove the changes to the replicate and split queues to different PR's The "primary" change in this PR is now to remove tracing from stores.go `Enqueue` and instead add it in the few places that case about it. If you ignore the test changes, this PR is fairly small now.

kvoli reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 20 of 24 files at r1, 1 of 1 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/132713#-:-O9V5kz79iHWOM5Hp1NX:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @angles-n-daemons, @jeffswenson, and @pav-kv)



<!-- Sent from Reviewable.io -->

andrewbaptist commented:
TFTR! 

bors r=kvoli
